### ISSUE TITLE
`Y034` emits spurious warnings for almost always intended usage in protocols

### ISSUE THREAD

**Issue by jonathandung** [PyCQA/flake8-pyi#539]
While type annotating functions such as `_operator.iadd`, I created protocols of the following form:

```python
@type_check_only
class _SupportsIAdd(Protocol[_T_contra, _T_co]):
    def __iadd__(self, other: _T_contra, /) -> _T_co: ...
```

`Y034` basically assumes that augmented assignment methods such as these should return `self` in the end, so the return type should be `Self` and the return annotation should not be generic. However, this assumption simply doesn't hold; while uncommon, classes may return a value of a different type in these methods (otherwise there would be no need to have a return value).

I propose to have `Y034` automatically silenced in protocol classes, for this reason.

See [typeshed #15611](https://github.com/python/typeshed/issues/15611).

**Comment by jonathandung** [comment]
Please note that I do not have much knowledge of the internals of this project, and will likely not be able to fix this issue.

**Comment by AlexWaygood** [comment]
This seems reasonable to me. I think it would be very trivial to implement, if you do fancy giving it a shot. We just want to quickly return `False` here if `class_ctx.is_protocol_class` (a property on `class_ctx` that already exists) evaluates to `True`: https://github.com/PyCQA/flake8-pyi/blob/e8a834b50f6dcff7f662c599ec47ed8247745ddd/flake8_pyi/visitor.py#L360-L367. We'd also need some new tests in https://github.com/PyCQA/flake8-pyi/blob/e8a834b50f6dcff7f662c599ec47ed8247745ddd/tests/classdefs.pyi, and a CHANGELOG entry

**Comment by jonathandung** [comment]
I think this counts as a breaking change. Should I include it in the changelog under the Breaking Changes heading, under Unreleased?
